### PR TITLE
Cleanup pypesto.profile

### PR DESCRIPTION
**Pull request by dweindl**
* documentation
* typehints
* simplify

### CODE REVIEW

File: pypesto/profile/profile_next_guess.py [hunk 1]
```diff
@@ -208,43 +207,47 @@ def clip_to_bounds(step_proposal):
...
-            # if we do simple extrapolation
-            x_step = x + step_length * delta_x_dir
+            return clip_to_bounds(x_step)
 
-        return clip_to_bounds(x_step)
+    else:
+        # if not, we do simple extrapolation
+        def par_extrapol(step_length):
+            x_step = x + step_length * delta_x_dir
+            return clip_to_bounds(x_step)
```
Comment (PaulJonasJost) [inline pypesto/profile/profile_next_guess.py:250]:
```suggestion
        return clip_to_bounds(x + step_length * delta_x_dir)
```
Is there a need for an extra function here?

File: pypesto/profile/validation_intervals.py [hunk 2]
```diff
@@ -50,34 +50,28 @@ def validation_profile_significance(
     problem_full_data:
         pypesto.problem, such that the objective is the
         negative-log-likelihood of the training and validation data set.
-
     result_training_data:
         result object from the fitting of the training data set only.
```
Comment (PaulJonasJost) [inline pypesto/profile/validation_intervals.py:54]:
```suggestion
        Result object from the fitting of the training data set only.
```
or adjust as other comment

**Review by PaulJonasJost — APPROVED** [review]

**Comment by PaulJonasJost** [comment]
thanks a lot for this :pray:

File: pypesto/profile/profile_next_guess.py (hunk 1 above)
Comment (dweindl) [reply pypesto/profile/profile_next_guess.py:250]: `par_extrapol` you mean? Yes, that is passed to `do_line_search` further down.

Comment (PaulJonasJost) [reply pypesto/profile/profile_next_guess.py:250]: ah sorry mixed up there 🙏🏼

**Pull request merged**